### PR TITLE
add travis build

### LINKED ISSUE #22: add travis build
probably using `tox`.

### PR DESCRIPTION
**Pull request by thijstriemstra**
using tox. fixes #22 

**still work in progress**

Getting following error when trying to build locally (ubuntu 16.04), expect the same to happen on travis.

```
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/usr/include/lirc/ -I/usr/local/include/python3.6m -c lirc/lirc.c -o build/temp.linux-x86_64-3.6/lirc/lirc.o
  gcc: error: lirc/lirc.c: No such file or directory
```

### CODE REVIEW

**Comment by tompreston** [comment]
Okay enabled.

**Comment by tompreston** [comment]
I was able to successfully build using the following Dockerfile:
```dockerfile
FROM debian:stable

ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
	autoconf automake build-essential libtool pkg-config
RUN apt-get update && apt-get install -y --no-install-recommends \
	liblirc-dev \
	python3-dev \
	python3-pip
RUN pip3 install --upgrade pip
RUN pip3 install Cython
WORKDIR /src

CMD make py3 && python3 setup.py install
```
With commands:
```
docker build . -t dev-python-lirc
docker run -v $PWD:/src -it dev-python-lirc
```

Maybe you're missing `liblirc-dev` which is required by:
```
$ grep "include_dirs" setup.py 
    include_dirs=['/usr/include/lirc/'],
```

**Comment by thijstriemstra** [comment]
Looks like [trusty](https://packages.ubuntu.com/search?suite=trusty&searchon=names&keywords=liblirc-dev) does not have `liblirc-dev`..

**Review by tompreston — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ matrix:
 
 install: 
   - pip install --ignore-installed --upgrade setuptools pip tox Cython
-  - wget https://downloads.sourceforge.net/project/lirc/LIRC/0.10.1/lirc-0.10.1.tar.bz2?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Flirc%2Ffiles%2FLIRC%2F0.10.1%2Flirc-0.10.1.tar.bz2%2Fdownload&ts=1534810232
+  - wget https://sourceforge.net/projects/lirc/files/latest/download
```
[the remark this answers is not in the document]
Comment (thijstriemstra) [reply .travis.yml]: tried that but didn't work either.

[the remark this answers is not in the document]
Comment (tompreston) [reply .travis.yml]: Now the download succeeds, but it looks like tar doesn't like the downloaded file. Can you see the Travis builds?

```
$ wget -O lirc-0.10.1.tar.bz2 https://sourceforge.net/projects/lirc/files/latest/download
--2018-08-21 11:37:47--  https://sourceforge.net/projects/lirc/files/latest/download
Resolving sourceforge.net (sourceforge.net)... 216.105.38.13
Connecting to sourceforge.net (sourceforge.net)|216.105.38.13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://downloads.sourceforge.net/project/lirc/LIRC/0.10.1/lirc-0.10.1.tar.bz2?r=&ts=1534851467&use_mirror=svwh [following]
--2018-08-21 11:37:47--  https://downloads.sourceforge.net/project/lirc/LIRC/0.10.1/lirc-0.10.1.tar.bz2?r=&ts=1534851467&use_mirror=svwh
Resolving downloads.sourceforge.net (downloads.sourceforge.net)... 216.105.38.13
Connecting to downloads.sourceforge.net (downloads.sourceforge.net)|216.105.38.13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://svwh.dl.sourceforge.net/project/lirc/LIRC/0.10.1/lirc-0.10.1.tar.bz2 [following]
--2018-08-21 11:37:47--  https://svwh.dl.sourceforge.net/project/lirc/LIRC/0.10.1/lirc-0.10.1.tar.bz2
Resolving svwh.dl.sourceforge.net (svwh.dl.sourceforge.net)... 72.5.72.15, 2606:c680:0:b:3830:34ff:fe66:6663
Connecting to svwh.dl.sourceforge.net (svwh.dl.sourceforge.net)|72.5.72.15|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2523226 (2.4M) [application/octet-stream]
Saving to: ‘lirc-0.10.1.tar.bz2’
100%[======================================>] 2,523,226   8.24MB/s   in 0.3s   
2018-08-21 11:37:48 (8.24 MB/s) - ‘lirc-0.10.1.tar.bz2’ saved [2523226/2523226]
0.01s$ tar -xzvf lirc-0.10.1.tar.bz2
gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
The command "tar -xzvf lirc-0.10.1.tar.bz2" failed and exited with 2 during .
```

**Comment by tompreston** [comment]
I'm not sure if you can see the travis build log:
```bash
$ pip install --ignore-installed --upgrade setuptools pip tox Cython PyYAML
Collecting setuptools
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/66/e8/570bb5ca88a8bcd2a1db9c6246bb66615750663ffaaeada95b04ffe74e12/setuptools-40.2.0-py2.py3-none-any.whl (568kB)
    100% |████████████████████████████████| 573kB 1.7MB/s 
Collecting pip
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 926kB/s 
Collecting tox
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/df/53/13660f04ef09ece4aefcce6b8f79c1586fc34dee1cbedd7c153e02f93489/tox-3.2.1-py2.py3-none-any.whl (62kB)
    100% |████████████████████████████████| 71kB 10.0MB/s 
Collecting Cython
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/9c/ae/be506d47f77fd31101a29eb740120f39050423f941cb0a461e4247330a47/Cython-0.28.5-cp27-cp27mu-manylinux1_x86_64.whl (3.3MB)
    100% |████████████████████████████████| 3.3MB 386kB/s 
Collecting PyYAML
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/9e/a3/1d13970c3f36777c583f136c136f804d70f500168edc1edea6daa7200769/PyYAML-3.13.tar.gz (270kB)
    100% |████████████████████████████████| 276kB 4.2MB/s 
Collecting six<2,>=1.0.0 (from tox)
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting pluggy<1,>=0.3.0 (from tox)
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/f5/f1/5a93c118663896d83f7bcbfb7f657ce1d0c0d617e6b4a443a53abcc658ca/pluggy-0.7.1-py2.py3-none-any.whl
Collecting virtualenv>=1.11.2 (from tox)
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/b6/30/96a02b2287098b23b875bc8c2f58071c35d2efe84f747b64d523721dc2b5/virtualenv-16.0.0-py2.py3-none-any.whl (1.9MB)
    100% |████████████████████████████████| 1.9MB 662kB/s 
Collecting py<2,>=1.4.17 (from tox)
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading https://files.pythonhosted.org/packages/f3/bd/83369ff2dee18f22f27d16b78dd651e8939825af5f8b0b83c38729069962/py-1.5.4-py2.py3-none-any.whl (83kB)
    100% |████████████████████████████████| 92kB 10.9MB/s 
Building wheels for collected packages: PyYAML
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /home/travis/.cache/pip/wheels/ad/da/0c/74eb680767247273e2cf2723482cb9c924fe70af57c334513f
Successfully built PyYAML
Installing collected packages: setuptools, pip, six, pluggy, virtualenv, py, tox, Cython, PyYAML
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/local/lib/python2.7/dist-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/local/lib/python2.7/dist-packages/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/usr/local/lib/python2.7/dist-packages/pip/wheel.py", line 323, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/usr/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/easy_install.pyc'
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
You are using pip version 9.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
The command "pip install --ignore-installed --upgrade setuptools pip tox Cython PyYAML" failed and exited with 2 during .
Your build has been stopped.
```

Shouldn't this be python3?
```
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/easy_install.pyc'
```

**Comment by thijstriemstra** [comment]
Yea will take a look when i have time.

**Comment by thijstriemstra** [comment]
What a PITA this turns out to be...

Still on https://travis-ci.com/tompreston/python-lirc/builds/82707667#L1829

**Comment by tompreston** [comment]
Hi @thijstriemstra, I noticed you pushed some changes recently. Did you get anywhere with this or can I close this PR? I intend to archive this repo soon. See https://github.com/tompreston/python-lirc/issues/28#issuecomment-712454662

**Pull request closed without merging**